ghcid-ng/src/ghci/mod.rs

```diff
@@ -272,7 +282,9 @@ impl Ghci {
                     needs_restart.push(path);
                 }
                 FileEvent::Modify(path) => {
-                    if self.modules.contains_source_path(&path)? {
+                    if self.modules.contains_source_path(&path)?
+                        || self.failed_modules.contains_source_path(&path)?
```

Comment:
Here's the actual bug fix. Don't `:add` modules that failed to compile. The rest of the PR is about populating the `failed_modules` set correctly.